postgresql/connection.go

```diff
@@ -0,0 +1,82 @@
+package postgresql
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+
+	"github.com/joho/godotenv"
+
+	_ "github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/dialers/postgres" // blank import
+)
+
+type StatsService struct {
+	db *sql.DB
+}
+
+func NewStatsService(config Config) (StatsService, error) {
+	dbURI := fmt.Sprintf("host=%s user=%s password=%s dbname=%s sslmode=disable",
+		config.Host,
+		config.User,
+		config.Password,
+		config.DBName)
+
+	db, err := sql.Open("cloudsqlpostgres", dbURI)
+	if err != nil {
+		return StatsService{}, ErrDatabaseConnection
+	}
+
+	err = db.Ping()
+	if err != nil {
+		return StatsService{}, ErrDatabasePing
+	}
+
+	db.SetMaxIdleConns(config.IdleConn)
+	db.SetMaxOpenConns(config.MaxConn)
+
+	return StatsService{db}, nil
+}
+
+type Config struct {
+	Host     string
+	User     string
+	Password string
+	DBName   string
+	IdleConn int
+	MaxConn  int
+}
+
+func GetConfigFromEnvVariables() (Config, error) {
+	var config Config
+
+	err := godotenv.Load(".env")
+	// no error returned here because .env is not deployed
+	if err != nil {
+		fmt.Println("no .env file found")
+	}
```

Comment:
[question] Why bother with an env file?

Reply:
It's just that `.env` is only meant to be used locally (when deployed env vars are directly injected), so this piece of code is only for local dev. Is it really necessary? We never run the worker locally, and if we come to need these variables for unit tests for instance we should set them in the test.

Reply:
Ah, indeed, I can remove it.